pkg/virtctl/guestfs/guestfs.go

```diff
@@ -334,22 +347,27 @@ func (client *K8sClient) getPodsForPVC(pvcName, ns string) ([]corev1.Pod, error)
 
 func createLibguestfsPod(pvc, image, cmd string, args []string, kvm, isBlock bool) *corev1.Pod {
 	var resources corev1.ResourceRequirements
-	var user, group int64
 	podName = fmt.Sprintf("%s-%s", podNamePrefix, pvc)
-	user = 0
-	group = 0
 	if kvm {
 		resources = corev1.ResourceRequirements{
 			Limits: corev1.ResourceList{
 				KvmDevice: resource.MustParse("1"),
 			},
 		}
 	}
+	var securityContext *corev1.PodSecurityContext
+	if uidInt != nil {
+		securityContext = &corev1.PodSecurityContext{
+			RunAsUser: uidInt,
+			FSGroup:   uidInt,
```

Comment:
Why do you need `FSGroup`? This will require privileged SCC.

Reply:
@xpivarc I'm using it because I want k8s to change the volume ownership and be sure I have the right permission to access the volume. From the comment in [k8s docs](https://pkg.go.dev/k8s.io/api/core/v1#PodSecurityContext) it seems necessary to me:

>  // A special supplemental group that applies to all containers in a pod.
> 	// Some volume types allow the Kubelet to change the ownership of that volume
> 	// to be owned by the pod:
> 	//
> 	// 1. The owning GID will be the FSGroup
> 	// 2. The setgid bit is set (new files created in the volume will be owned by FSGroup)
> 	// 3. The permission bits are OR'd with rw-rw----
> 	//
> 	// If unset, the Kubelet will not modify the ownership and permissions of any volume.
> 	// Note that this field cannot be set when spec.os.name is windows.
> 	// +optional
> 	FSGroup *[int64](https://pkg.go.dev/builtin#int64) `json:"fsGroup,omitempty" protobuf:"varint,5,opt,name=fsGroup"`

The ability to set the uid is to be able to use older libguestfs-tool images. Unfortunately, the file there is owned by root and  you need to run with uid=0. If you have any better suggestion is very welcome. I haven't found a better way to guarantee some degree of backward compatibility

Reply:
If the user doesn't have the SSC to set it then it should rely on the defaults and not using that flag

Reply:
Understand, so I would just check if the requested user is root as root should not need the `fsGroup`.